pkg/objectfile/pool.go

```diff
@@ -92,40 +107,61 @@ type Pool struct {
 	c       burrow.Cache
 }
 
-func NewPool(logger log.Logger, reg prometheus.Registerer, size int) *Pool {
+const keepAliveProfileCycle = 30
+
+func NewPool(logger log.Logger, reg prometheus.Registerer, profilingDuration time.Duration) *Pool {
 	return &Pool{
 		metrics: newMetrics(reg),
 		c: burrow.New(
-			// An ideal size for the pool needs to be determined.
-			// A lesser size will cause premature closing of files.
-			burrow.WithMaximumSize(size),
-			burrow.WithRemovalListener(onRemoval(log.With(logger, "component", "objectfile_pool"))),
+			burrow.WithExpireAfterAccess(keepAliveProfileCycle*profilingDuration),
 			burrow.WithStatsCounter(cache.NewBurrowStatsCounter(logger, reg, "objectfile")),
 		),
 	}
 }
 
+// Get returns the ObjectFile reference for the given buildID if it exists in the cache.
+func (p *Pool) Get(buildID string) (*ObjectFile, error) {
```

Comment:
I think it's something evolved in a long period of time. I don't remember the first reason. It would be an easy change and makes perfect change. I'll do it.

Reply:
I remembered the reason, the same object file can be mapped different processes. The unique identifier is the build id. However, this can be solved with an additional path to build id mapping. Which is cheaper then opening the file and reading the build id.

Reply:
https://github.com/parca-dev/parca-agent/pull/1689/commits/4aa7d031ac282044d43238d3256922d622805ed4